rna/input_output.py

```diff
@@ -0,0 +1,250 @@
+"""
+Reads and manipulates datasets.
+"""
+
+from collections import defaultdict
+
+import numpy as np
+import pandas as pd
+
+from sklearn.preprocessing import LabelEncoder
+
+
+# TODO: include else option when '_rv' not in filename
+# TODO: imports file that contains 4 rv per sample without rv's connected to
+def read_df(filename, binarize, number_of_replicates=1):
+    """
+    Reads in an xls file as a dataframe, replacing NA and binarizing if required.
+    Returns the original dataframe and a separate list of indices that are the
+    replicate numbers that belong to each sample.
+
+    Note that this function only works if '_rv' is in the filename.
+
+    :param filename: name of file to read in
+    :param binarize: whether to binarize - use a cutoff value to convert to 0/1
+    :return: (dataframe, list of indices)
+    """
+    if '_rv' in filename:
+        # then sure that it includes 'replicate_values'
+        df_rv = pd.read_excel(filename, delimiter=';', index_col=0)
+        df = df_rv.loc[:, (df_rv.columns.values[:-1])]
+        rv = df_rv[['replicate_value']]
+        df.fillna(0, inplace=True)
+        if binarize:
+            df = 1 * (df > 150)
+        return df, rv
+
+    else:
+        # TODO: What type of data is to be expected?
+        # TODO: Add replicate numbers manually
+        # then 'replicate_values' not included, so
+        # assume that all samples have 4 replicates
+        df = pd.read_excel(filename, delimiter=';', index_col=0)
+        df.fillna(0, inplace=True)
+        if binarize:
+            df = 1 * (df > 150)
+
+        try:
+            rv = np.zeros((df.shape[0], 1))
```

Comment:
ik zou hier een harde foutmelding geven als het totaal aantal regels geen veelvoud is van number_of_replicates - want dan is er ergens iets aan de hand!